pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -99,6 +104,33 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	if err != nil {
 		return errors.Wrap(err, "could not create watch on WindowsMachineConfig objects")
 	}
+	// Watch for the machine objects
+	windowsOSLabel := "node.openshift.io/os_id"
+	predicateFilter := predicate.Funcs{
+		// ignore create event for all machines as wmco should for machine getting provisioned
+		CreateFunc: func(e event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			labels := e.MetaNew.GetLabels()
+			if _, ok := labels[windowsOSLabel]; ok {
```

Comment:
Is it enough if the label exist especially if we go with `machine.openshift.io/os_id`?